photon-server/src/main/java/org/photonvision/server/Server.java

```diff
@@ -27,45 +28,45 @@ public class Server {
-                                                        .add(ctx.req.getMethod())
+                                                        .add(ctx.req().getMethod())
                                                         .add("from endpoint")
                                                         .add(ctx.path())
                                                         .add("for host")
-                                                        .add(ctx.req.getRemoteHost())
+                                                        .add(ctx.req().getRemoteHost())
                                                         .add("in")
                                                         .add(ms.toString())
                                                         .add("ms");
 
                                         logger.debug(joiner.toString());
                                     });
-
-                            config.wsLogger(
-                                    ws ->
-                                            ws.onMessage(
-                                                    ctx -> logger.debug("Got WebSockets message: " + ctx.message())));
-
-                            config.wsLogger(
-                                    ws ->
-                                            ws.onBinaryMessage(
-                                                    ctx ->
-                                                            logger.trace(
-                                                                    () -> {
-                                                                        var insa = ctx.session.getRemote().getInetSocketAddress();
-                                                                        var host = insa.getAddress().toString() + ":" + insa.getPort();
-                                                                        return "Got WebSockets binary message from host " + host;
-                                                                    })));
+                            javalinConfig.requestLogger.ws(
+                                    ws -> {
+                                        ws.onMessage(ctx -> logger.debug("Got WebSockets message: " + ctx.message()));
+                                        ws.onBinaryMessage(
+                                                ctx ->
+                                                        logger.trace(
+                                                                () -> {
+                                                                    var remote = (InetSocketAddress) ctx.session.getRemoteAddress();
+                                                                    var host =
+                                                                            remote.getAddress().toString() + ":" + remote.getPort();
+                                                                    return "Got WebSockets binary message from host: " + host;
```

Comment:
This might get spammy, but should be fine since messages going this direction are rare

Reply:
And it was already like this